content/draft/general/keyboard_shortcuts.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Keyboard shortcuts spec"
+description: ""
+date: 2020-12-01T08:48:57+00:00
+lastmod: 2020-12-01T08:48:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "general"
+weight: 100
+toc: true
+---
+
+# Keyboard Shortcuts spec
+
+## Motivation
+
+Users often use keyboard shortcuts to more effectively navigate applications. This spec aims to document the keyboard shortcuts available as well as any preconditions.
+
+## Designs
+
+* [designs](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=355%3A82493)
+
+## Functional Requirements
+
+### Key definition
+
+The requirements will use the keys `CMD` and `OPT` which are defined for each platform as follows:
+
+- `CMD`
+    - macOS: `⌘` or `CMD`
+    - Windows: `Ctrl`
+    - Linux: `Ctrl`
+- `OPT`
+    - macOS: `⌥` or `OPT`
+    - Windows: `Alt`
+    - Linux: `Alt`
+
+### Keyboard shortcuts help
+
+- `CMD + /` MUST open a modal listing available key shortcuts
+- The Modal listing available keyshortcuts MUST display the correct keys depending on the platform, for example `CMD + C` on macOS and `Ctrl + C` on Windows.
+
+### Keyboard shortcuts
+
+#### Message focus
+
+- `TAB` - MUST trigger focus of the last message
+- `UP` - MUST focus on the previous message
+    - requirement: there is already a message in focus
+- `DOWN` - MUST focus on the previous message
+    - requirement: there is already a message in focus
+- `ESCAPE`:
+    - if a message is already in focus, it MUST be removed from focus and the input box MUST be focused instead
+    - if input box is already in focus then focus MUST BE removed completly
+    - if there is nothing in focus then the input box MUST be focused
+
+#### Message selection shortcuts
+
+requirement: there is already a message in focus
+
+- `E` MUST edit the currently focused message
+    - requirement: the message must have been sent by the current user
+- `Backspace` MUST delete the currently focused message
+    - requirement: the message must have been sent by the current user
+- `P` pins the currently focused message
+    - requirement: the current user must be an admin in a community or a group chat
+- `+` react to currently focused message
+- `R` - reply to currently focused message
+- `CMD + C` - copy currently focused message
+- `OPT + ENTER` - mark currently focused message as read
+
+#### Navigation shortcuts
+
+- `OPT + ENTER + UP` - MUST select the previous community, if there is no previous community then it MUST select the last community
+- `OPT + ENTER + DOWN` - MUST select the next community, if there is no next community then it MUST select the first community
+- `OPT + UP` - MUST select the previous channel, if there is no previous channel then it MUST select the last channel
+- `OPT + DOWN` - MUST select the next channel, if there is no next channel then it MUST select the first channel
+- `CMD + [` - MUST Navigate backward in page history
```

Comment:
what is the difference between this and the `PAGE UP` short cut below?

Reply:
What is a page history in this context? What's the context anyways? Are we in chat/message view again? If yes, please specify

Reply:
> 

I think the difference is "OPT + UP" and "OPT + DOWN" cycle between channels, while "PAGE UP" and "PAGE DOWN" scroll the main chat view

Reply:
@PascalPrecht all of these shortcuts are basically copied from Discord (because we want any user who is coming from Discord to be able to use all the Discord shortcuts they have already learnt), so the best way to answer questions about what these shortcuts are supposed to do is to try them in Discord ;-)